src/plugins/intel_npu/src/plugin/npuw/llm_infer_request.cpp

```diff
@@ -0,0 +1,239 @@
+                                               ov::SoPtr<ov::ITensor> position_ids) {
+    LOG_DEBUG("LLMInferRequest::infer_generate()");
+    LOG_BLOCK();
+
+    // NB: KV-cache is full, further generation is impossible
+    if (m_kvcache_desc.num_stored_tokens == m_kvcache_desc.total_size) {
+        LOG_DEBUG("KV-Cache is full");
+        return;
+    }
+
+    if (m_need_copy_kvcache) {
+        LOG_DEBUG("Copying kv-cache from prefill to generate model.");
+        const auto kStartOutputKVCacheLayers = 1u;
+        const auto& kvcache_compiled = m_kvcache_request->get_compiled_model();
+        for (int i = 0; i < kvcache_compiled->outputs().size() - 1; ++i) {
+            const auto& output_name = kvcache_compiled->outputs()[kStartOutputKVCacheLayers + i].get_any_name();
+            auto prefill_out_tensor = ov::make_tensor(m_prefill_request->get_tensor(m_prefill_out_ports.at(output_name)));
+
+            const auto& input_name = std::regex_replace(output_name, std::regex("present"), "past_key_values");
+            auto kvcache_in_tensor = ov::make_tensor(m_kvcache_request->get_tensor(m_kvcache_in_ports.at(input_name)));
+            fill_tensor<ov::float16>(kvcache_in_tensor, 0);
+
+            auto prefill_out_slice = make_tensor_slice(
+                prefill_out_tensor, m_kvcache_desc.dim,
+                m_kvcache_desc.max_prompt_size - m_kvcache_desc.num_stored_tokens, m_kvcache_desc.max_prompt_size
+            );
+
+            auto kvcache_in_slice = make_tensor_slice(
+                kvcache_in_tensor, m_kvcache_desc.dim, 0u, m_kvcache_desc.num_stored_tokens
+            );
+            prefill_out_slice.copy_to(kvcache_in_slice);
+            // FIXME: copy_to doesn't work for 31 key/values layers
+            //        But it might be more optimal copy solution here.
+            // copy_to(prefill_out_tensor, kvcache_in_tensor, 2,
+            //         m_kvcache_desc.max_prompt_size - m_kvcache_desc.num_stored_tokens,
+            //         0, m_kvcache_desc.num_stored_tokens);
+        }
+        LOG_DEBUG("LLMInferRequest::infer_generate(): Prepare attention mask pattern.");
+        auto* attention_mask_data = m_kvcache_request->get_tensor(
+            m_kvcache_in_ports.at("attention_mask"))->data<int64_t>();
+        attention_mask_data[m_kvcache_desc.total_size - 1] = 1;
+
+        m_need_copy_kvcache = false;
+    }
+
+    auto kv_input_ids = m_kvcache_request->get_tensor(m_kvcache_in_ports.at("input_ids"));
+    std::copy_n(input_ids->data<int64_t>(), input_ids->get_size(), kv_input_ids->data<int64_t>());
+
+    auto kv_attn_mask = m_kvcache_request->get_tensor(m_kvcache_in_ports.at("attention_mask"));
+    std::copy_n(attention_mask->data<int64_t>(), attention_mask->get_size(), kv_attn_mask->data<int64_t>());
+
+    auto kv_pos_ids = m_kvcache_request->get_tensor(m_kvcache_in_ports.at("position_ids"));
+    std::copy_n(position_ids->data<int64_t>(), position_ids->get_size(), kv_pos_ids->data<int64_t>());
```

Comment:
add a FIXME here please; these tensors should be shared between the parent & child models

Reply:
Added